variation/tokenize.py

```diff
@@ -0,0 +1,98 @@
+"""A module for tokenization."""
+from typing import Iterable, List
+
+from variation.schemas.token_response_schema import Token, TokenType
+from variation.tokenizers import (
+    HGVS,
+    CdnaDeletion,
+    CdnaDelIns,
+    CdnaGenomicReferenceAgree,
+    CdnaInsertion,
+    CdnaSubstitution,
+    FreeTextCategorical,
+    GeneSymbol,
+    GenomicDeletion,
+    GenomicDelIns,
+    GenomicDuplication,
+    GenomicInsertion,
+    GenomicSubstitution,
+    GnomadVCF,
+    ProteinDeletion,
+    ProteinDelIns,
+    ProteinInsertion,
+    ProteinReferenceAgree,
+    ProteinSubstitution,
+)
+
+r"(\((\?|d+)_(\?|\d+)\))_(\((\?|\d+)_(\?|\d+)\))dup"
+
+
+class Tokenize:
+    """The tokenize class."""
+
+    def __init__(self, gene_symbol: GeneSymbol) -> None:
+        """Initialize the tokenize class."""
+        self.gene_symbol = gene_symbol
+        self.tokenizers = [
+            HGVS(),
+            GnomadVCF(),
+            self.gene_symbol,
+            FreeTextCategorical(),
+            # Substitution
+            ProteinSubstitution(),
+            GenomicSubstitution(),
+            CdnaSubstitution(),
+            # Reference Agree
+            ProteinReferenceAgree(),
+            CdnaGenomicReferenceAgree(),
+            # Delins
+            ProteinDelIns(),
+            CdnaDelIns(),
+            GenomicDelIns(),
+            # Deletion
+            ProteinDeletion(),
+            CdnaDeletion(),
+            GenomicDeletion(),
+            # Insertion
+            ProteinInsertion(),
+            CdnaInsertion(),
+            GenomicInsertion(),
+            # Duplication
+            GenomicDuplication(),
+        ]
+
+    def perform(self, search_string: str, warnings: List[str]) -> Iterable[Token]:
```

Comment:
any reason this is left as an `Iterable`? It looks like it has to be a `List`

Reply:
Nope, forgot to look at previous return types to update